flash/data/process.py

```diff
@@ -101,6 +101,151 @@ class PreprocessState:
 
 
 class Preprocess(Properties, torch.nn.Module):
+    """
+    The :class:`~flash.data.process.Preprocess` is used to encapsulate
+    all the processing data loading logic up to the model.
+
+    It is particularly relevant when you want to provide ready-to-production implementation which works
+    with 4 different stages: ``training``, ``validation``, ``test` and inference(``predicting``).
+
+    The :class:`~flash.data.process.Preprocess` supports the following hooks:
+
+        - ``load_data``: Function to receiving some metadata to generate a Mapping from.
```

Comment:
I think I would remove the bullet points. It looks too thin in the rendered docs

Reply:
Also can we be more noob friendly than `Function to receiving some metadata to generate a Mapping from`?

Reply:
I don't understand this. Let's try to use verbs to describe hooks. maybe
"Given x, returns Y". or "Given a pointer to the data, returns an iterable collection of the data samples and labels"?


If we can't exaplin it, people will def not know how to use